dbms/src/Storages/DeltaMerge/RowKeyFilter.h

```diff
@@ -89,9 +89,10 @@ inline Block filterSorted(const RowKeyRanges & rowkey_ranges, Block && block, si
         return {};
 
     std::vector<std::pair<size_t, size_t>> offset_and_limits;
-    for (auto rowkey_range : rowkey_ranges)
+    offset_and_limits.reserve(rowkey_ranges.size());
+    for (const auto & rowkey_range : rowkey_ranges)
     {
-        offset_and_limits.emplace_back(getPosRangeOfSorted(rowkey_range, block.getByPosition(handle_pos).column, 0, block.rows()));
+        offset_and_limits.emplace_back(getPosRangeOfSorted(rowkey_range, block.safeGetByPosition(handle_pos).column, 0, block.rows()));
```

Comment:
I think we can add some comments in `filterSorted`, or add some pre-checks lines under debug mode to ensure `handle_pos` is a valid index. And use getByPosition to get a better performance in release mode.

Reply:
all fixed.